piet-coregraphics/src/ct_helpers.rs

```diff
@@ -34,22 +40,98 @@ pub(crate) struct Frame(pub(crate) CTFrame);
 #[derive(Debug, Clone)]
 pub(crate) struct Line<'a>(pub(crate) Cow<'a, CTLine>);
 
+pub enum __CTParagraphStyle {}
+type CTParagraphStyleRef = *const __CTParagraphStyle;
+
+declare_TCFType!(CTParagraphStyle, CTParagraphStyleRef);
+impl_TCFType!(
+    CTParagraphStyle,
+    CTParagraphStyleRef,
+    CTParagraphStyleGetTypeID
+);
+
+#[repr(u32)]
+enum CTParagraphStyleSpecifier {
+    Alignment = 0,
+    //FirstLineHeadIndent = 1,
+    //HeadIndent = 2,
+    //TailIndent = 3,
+    //TabStops = 4,
+    //TabInterval = 5,
+    //LineBreakMode = 6,
+    // there are many more of these
+}
+
+#[repr(u8)]
```

Comment:
This all seems a bit strange, but I'll trust that you researched and tested this.

Reply:
indeed. Had a few choices here, this feels the most true to the semantics of CoreText. The `CTParagraphStyleSetting` business is pretty hairy API; this is the bit I'd got hung up on, because the field ordering in the docs is different from the actual layout of the struct, which I eventually found in a header.